packages/firestore/test/util/equality_matcher.ts

```diff
@@ -54,26 +54,30 @@ function customDeepEqual(left, right) {
 }
 
 export function addEqualityMatcher() {
-  let originalFunction;
-  beforeEach(() => {
+  let isActive = true;
+
+  before(() => {
     use((chai, utils) => {
       const Assertion = chai.Assertion;
 
       const assertEql = _super => {
-        originalFunction = originalFunction || _super;
         return function(...args) {
-          const [right, msg] = args;
-          utils.flag(this, 'message', msg);
-          const left = utils.flag(this, 'object');
+          if (isActive) {
+            const [right, msg] = args;
+            utils.flag(this, 'message', msg);
+            const left = utils.flag(this, 'object');
 
-          this.assert(
-            customDeepEqual(left, right),
-            'expected #{act} to roughly deeply equal #{exp}',
-            'expected #{act} to not roughly deeply equal #{exp}',
-            left,
-            right,
-            true
-          );
+            this.assert(
+              customDeepEqual(left, right),
+              'expected #{act} to roughly deeply equal #{exp}',
+              'expected #{act} to not roughly deeply equal #{exp}',
+              left,
+              right,
+              true
+            );
+          } else {
+            _super.apply(this, args);
+          }
```

Comment:
FWIW, I suspect this code works but is a bit bonkers.  We'll end up overwriting the method multiple times and I think each _super will point to the previous method, so as this code executes repeatedly you'll end up with a chain of _super's pointing to each other and so _super() will call _super() will call _super() until you finally get back to the original function which won't call _super()...  To make things extra bonkers, there will be multiple instances of the isActive variable (one for each time addEqualityMatcher() has been called since the tests started)...

But I don't really care. :-)

Reply:
I mostly uploaded this to give Josh a chance to look at this... but now that I have your attention:

- The existing code never actually performed any cleanup and even if it had, this cleanup would only have worked if deepEqual had ever been called. 
- With the new cleanup, some tests started failing that compared Map values directly. I added the equality helper to these tests.
- I removed some of the 'bonkersness' by keepin the original Function around. This should reduce the chain of _super._super._super calls.